lib/openhab/dsl/monkey_patch/types/on_off_type.rb

```diff
@@ -33,16 +33,15 @@ def !
           #  nil if object cannot be compared
           #
           def ===(other)
-            # A case statement here causes and infinite loop
-            # rubocop:disable Style/CaseLikeIf
-            if self == ON
-              other.on? if other.respond_to? :on?
+            if other.instance_of?(self.class)
```

Comment:
The other command types use is_a? instead of instance_of?.  They should probably be uniform

Reply:
I initially used is_a? but rubocop rewrote it as instance_of?